crates/planner/tests/test.rs

```diff
@@ -49,3 +53,63 @@ fn test() {
         }
     }
 }
+
+#[test]
+fn test_federation() {
+    tracing_subscriber::fmt()
+        .with_env_filter(
+            EnvFilter::try_from_default_env()
+                .or_else(|_| EnvFilter::try_new("debug"))
+                .unwrap(),
+        )
+        .init();
+    let collections_service_document = parser::parse_schema(include_str!("collections.graphql")).unwrap();
+    let collectibles_service_document = parser::parse_schema(include_str!("collectibles.graphql")).unwrap();
+    let data = fs::read_to_string("./tests/federation.text").unwrap();
+    let mut s = data.split("---");
+    let graphql = s.next().unwrap();
+    let variables = s.next().unwrap();
+    let planner_json = s.next().unwrap();
+    let schema = ComposedSchema::combine([
+        ("collectibles".to_string(), collectibles_service_document.clone()),
+        ("collections".to_string(), collections_service_document.clone()),
+    ])
+    .unwrap();
+    let reverse_order_schema = ComposedSchema::combine([
+        ("collections".to_string(), collections_service_document),
+        ("collectibles".to_string(), collectibles_service_document),
+    ])
+    .unwrap();
+    let document = parser::parse_query(graphql).unwrap();
+    let reverse_document = parser::parse_query(graphql).unwrap();
+    assert_eq!(schema.query_type, reverse_order_schema.query_type);
+    assert_eq!(schema.mutation_type, reverse_order_schema.mutation_type);
+    assert_eq!(schema.subscription_type, reverse_order_schema.subscription_type);
+    // assert_eq!(schema.types, reverse_order_schema.types);
+    assert_eq!(schema.directives, reverse_order_schema.directives);
+    assert_eq!(
+        serde_json::to_value(&document).unwrap(),
+        serde_json::to_value(&reverse_document).unwrap()
```

Comment:
Isn't this unnecessary as they are the exact same things just with different name?